src/main/haskell/kore/src/Kore/Step/Simplification/DomainValue.hs

```diff
@@ -48,3 +74,110 @@ simplify dv =
         ]
     , SimplificationProof
     )
+-}
+
+simplifyBuiltinDomain
+    :: ( Eq (variable Object), Show (variable Object)
+       , Given (SortTools Object)
+       , SortedVariable variable
+       )
+    => Sort Object
+    -> BuiltinDomain (OrOfExpandedPattern Object variable)
+    -> OrOfExpandedPattern Object variable
+simplifyBuiltinDomain domainValueSort =
+    \case
+        BuiltinDomainPattern pat ->
+            OrOfExpandedPattern.make
+            [
+                (ExpandedPattern.fromPurePattern . asPurePattern)
+                (DomainValuePattern DomainValue
+                    { domainValueSort
+                    , domainValueChild = BuiltinDomainPattern pat
+                    }
+                )
+            ]
+        BuiltinDomainMap _map -> do
+            (_map, predicate, substitution) <-
+                Map.foldlWithKey' simplifyBuiltinDomainMapElement
+                    (pure (Map.empty, Predicate.makeTruePredicate, mempty))
+                    _map
+            let term =
+                    (asPurePattern . DomainValuePattern)
+                    DomainValue
+                        { domainValueSort
+                        , domainValueChild = BuiltinDomainMap _map
+                        }
+            OrOfExpandedPattern.make
```

Comment:
The standard for the simplify functions which are similar to the ones in this file (Application and And) is that they `mergePredicatesAndSubstitutions` which also does some normalization and simplification. Arguably, we may want to change that at some point, and maybe only normalize them when that's actually needed, but, for now, it's likely that there are edge cases where simplification will not work properly without it.